test/unit/attributes.js

```diff
@@ -1107,6 +1107,71 @@ QUnit.test( "val(select) after form.reset() (Bug #2551)", function( assert ) {
 	jQuery( "#kk" ).remove();
 } );
 
+QUnit.test( "select.val(space characters) (gh-2978)", function( assert ) {
```

Comment:
That said, I'll land this for now so we can get 1.12.2/2.2.2 out, but I'm still interested in what you're proposing.